src/v/cluster/types.h

```diff
@@ -4142,6 +4142,19 @@ enum class reconfiguration_state {
     force_cancelled = 3
 };
 
+inline bool is_cancelled_state(reconfiguration_state rs) {
+    switch (rs) {
+    case reconfiguration_state::in_progress:
+    case reconfiguration_state::force_update:
+        return false;
+    case reconfiguration_state::cancelled:
+    case reconfiguration_state::force_cancelled:
+        return true;
+    default:
+        __builtin_unreachable();
```

Comment:
if the default case should never happen, but the cases aren't a covering set, then i think assertion is better here. otherwise, if the cases are a covering set, then you can omit the default and put the builtin_unreachable outside the switch statement to suppress the warning on gcc.